### PR TITLE
python310Packages.lifelines: init at 0.27.4

### PR DESCRIPTION
**Pull request by swflint**
###### Description of changes

`lifelines` is a python library used for survival analysis.  It requires a couple of additional packages which are not yet available in nixpkgs, and so these are packaged as well.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
Please add pythonImportChecks to everything. Also are there tests we can run?

**Comment by swflint** [comment]
@SuperSandro2000 Thanks for the feedback!  I've gone ahead and made those changes, and hopefully will have formulaic fixed up shortly.

File: pkgs/development/python-modules/formulaic/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    leaveDotGit = true;
+  };
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  nativeBuildInputs = [
+    git
+  ];
+
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/formulaic/default.nix:38]:
```suggestion
```
Do we really need git? Also hatchling is run at build time so we need it native.

Comment (swflint) [reply pkgs/development/python-modules/formulaic/default.nix:38]: With SETUPTOOLS_SCM_PRETEND_VERSION set, git is no longer necessary.

Thanks.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SamLukeYes** [comment]
Is this PR still active? A new version `0.27.4` of `lifelines` is available.

**Comment by SuperSandro2000** [comment]
I'll push the update and another minor fix soon.

**Comment by SamLukeYes** [comment]
Result of `nixpkgs-review pr 193990` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.autograd-gamma</li>
    <li>python310Packages.formulaic</li>
    <li>python310Packages.interface-meta</li>
    <li>python310Packages.lifelines</li>
    <li>python311Packages.autograd-gamma</li>
    <li>python311Packages.formulaic</li>
    <li>python311Packages.interface-meta</li>
    <li>python311Packages.lifelines</li>
  </ul>
</details>

**Comment by SamLukeYes** [comment]
I don't know why `nixpkgs-review` refuses to build it, but I've built `python3.11-lifelines` successfully, despite some warnings in tests.

Result of `nix log /nix/store/7c3p55vbgmmgd5jbs25blpr2g
9gb04gn-python3.11-lifelines-0.27.4.drv`: [python3.11-lifelines-0.27.4.log](https://github.com/NixOS/nixpkgs/files/10862052/python3.11-lifelines-0.27.4.log)

**Pull request merged**